### PR TITLE
 When changing the type of a metric with configured sources, don't re…

### LINKED ISSUE #4443: Changing the type of a metric deletes the source, even if the source supports both the old and new type
**Describe the bug**
Changing the type of a metric deletes the source, even if the source supports both the old and new type.

**Expected behavior**
The sources that support both the old and the new metric type should not be deleted.

**Your environment (please complete the following information):**
 - Quality-time version: v4.3.0

### PR DESCRIPTION
**Pull request by fniessink**
…move the sources that support both the old and the new metric type. Fixes #4443.

### CODE REVIEW

File: docs/src/usage.md [hunk 1]
```diff
@@ -205,7 +205,11 @@ After you've added a metric, the metric is visible in the subject's metric table
 :class: only-dark
 ```
 
-The first parameter is the "Metric type". The metric type determines what gets measured. When you change the metric type, the sources you can select in the "Sources" tab change accordingly. By default, the name of the metric is equal to its type, "Accessibility violations" in the example above, but you can change the metric name using the "Metric name" field.
+The first parameter is the "Metric type". The metric type determines what gets measured. By default, the name of the metric is equal to its type, "Accessibility violations" in the example above, but you can change the metric name using the "Metric name" field. When you change the metric type, the sources you can select in the "Sources" tab change accordingly.
+
+```{warning}
+If you change the type of a metric that has sources configured, sources that do not support the new metric type will be removed.
```
Comment (wkoot) [inline docs/src/usage.md:211]: The UI should display a warning if this is the case

File: components/external_server/tests/routes/test_metric.py [hunk 2]
```diff
@@ -114,11 +115,15 @@ def test_post_metric_name(self, request):
         )
 
     def test_post_metric_type(self, request):
-        """Test that the metric type can be changed."""
+        """Test that the metric type can be changed and that sources that support the new type are not removed."""
```
Comment (wkoot) [inline components/external_server/tests/routes/test_metric.py:119]: Should there not be a test that verifies changing `old_type` of `source_type2` will yield `new_type` without sources?

File: docs/src/usage.md (hunk 1 above)
Comment (fniessink) [reply docs/src/usage.md:211]: We have a separate story to address that: #4444.

File: components/external_server/tests/routes/test_metric.py (hunk 2 above)
Comment (fniessink) [reply components/external_server/tests/routes/test_metric.py:119]: The metric has two sources, one of which supports the new metric type, the other doesn't. So when the metric type is changed from old type to new type, the source that supports the new type is retained and the other is removed. I think that covers all scenario's.

**Pull request merged**